Backend/FcEvaluate.hs

```diff
@@ -46,16 +48,20 @@ smallStep (FcTmCaseFc e1@(FcTmERROR {}) _) = pure $ Just e1
 -- Useful cases (substitution)
```

Comment:
Every case that (a) is not a congruence (so that you don't count doubles) but )b) does not return `Nothing` (which means a step is taken) should call `increment`. This means that you forgot to call it on three cases:
```
-- Error propagation
smallStep (FcTmApp    e1@(FcTmERROR {}) _) = pure $ Just e1
smallStep (FcTmTyApp  e1@(FcTmERROR {}) _) = pure $ Just e1
smallStep (FcTmCaseFc e1@(FcTmERROR {}) _) = pure $ Just e1
```